processors/src/main/java/io/opentelemetry/contrib/interceptor/common/ComposableInterceptor.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.interceptor.common;
+
+import io.opentelemetry.contrib.interceptor.api.Interceptor;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Allows to run an item through a list of interceptors in the order they were added. */
+public final class ComposableInterceptor<T> implements Interceptor<T> {
+  private final List<Interceptor<T>> interceptors = new ArrayList<>();
```

Comment:
Typically, when using the `BatchSpanProcessor` sending the telemetry data happens on a separate thread; consider using `CopyOnWriteArrayList` instead
```suggestion
  private final List<Interceptor<T>> interceptors = new CopyOnWriteArrayList<>();
```